src/components/QuoteApi.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from 'react';
+
+const apiKey = new Headers();
+apiKey.append('X-Api-Key', 'fi23b02FxA5RuBZcQMwheA==1pRuDF9DyRGAqaht');
+const apiSite = 'https://api.api-ninjas.com/v/quotes?category=happiness';
+
+export default function QuoteApi() {
+  const [quote, setQuote] = useState([]);
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [quoteLoading, setQuoteLoading] = useState(false);
+
+  useEffect(() => {
+    setQuoteLoading(true);
+    const fetchQuotes = async () => {
+      try {
+        const response = await fetch(apiSite, {
+          method: 'GET',
+          headers: apiKey,
+        });
+
+        if (!response.ok) {
+          setErrorMessage('Network response was not ok.');
+        }
+
+        const fetchedQuotes = await response.json();
+        setQuote(fetchedQuotes);
+      } catch (error) {
+        setQuoteLoading(false);
+        setErrorMessage(`Error fetching data: ${error.message}`);
+      }
+    };
+
+    fetchQuotes();
+  }, []);
+
+  return (
+    <div>
+      <div className="flex flex-grow-0 items-center ml-10 justify-center max-w-1/4 rounded-md text-center w-4/6 bg-slate-600 border-solid border-black border-2">
+        <div className="bg-blue-400 shadow-2xl rounded-md p-6 max-w-1/4 text-purple-700 font-bold text-center text-2xl">
+          {quoteLoading ? (
+            <ul>
+              {quote.map((item) => (
+                <li key={`${item.quote}-${item.author}`}>
+                  Quote:
+                  <br />
+                  {item.quote}
+                  <br />
+                  <br />
+                  Author:
+                  {' '}
+                  {item.author}
+                </li>
+              ))}
+            </ul>
+          ) : (
+            <p>{errorMessage}</p>
+          )}
```

Comment:
- [ ] Good job working on the loading as highlighted by the previous reviewer. But however, you're to show some text when loading is true or show an error message when there is an error, and if no error and not loading, then the quotes are rendered. However, the quotes data are not being fetched at the moment due to a CORS error as shown in the screenshot below. Kindly work on that so the quotes are fetched and rendered successfully 👍🏽

<img width="1813" alt="Screenshot 2023-07-22 at 8 20 11 PM" src="https://github.com/Kanyorok/mathmagician/assets/50941074/fba1652e-9f84-4d12-9c80-2903199c93de">

- [ ] To show the loading or error or data, kindly see a sample code below to help guide you.


```jsx
if (errorMessage) return <div>Something went wrong!</div>

  if (quoteLoading) return <div>Loading...</div>

  return (
    <ul>
      {quote.map(({ author, quote }) => (
        <li key={author}>{quote}</li>
      ))}
    </ul>
  )
```